PROTO_tests/tests/dataframe_test.py

```diff
@@ -0,0 +1,704 @@
+        pd_df.reset_index(drop=True, inplace=True)
+
+        assert_frame_equal(pddf_drop, df_drop.to_pandas())
+
+        # verify that index keys must be ints
+        with pytest.raises(TypeError):
+            df.drop("index")
+
+        # verify axis can only be 0 or 1
+        with pytest.raises(ValueError):
+            df.drop("amount", 15)
+
+    def test_drop_duplicates(self):
+        df = self.build_ak_df_duplicates()
+        ref_df = self.build_pd_df_duplicates()
+
+        dedup = df.drop_duplicates()
+        dedup_pd = ref_df.drop_duplicates()
+        # pandas retains original indexes when dropping dups, need to reset to line up with arkouda
+        dedup_pd.reset_index(drop=True, inplace=True)
+
+        dedup_test = dedup.to_pandas().sort_values("userName").reset_index(drop=True)
+        dedup_pd_test = dedup_pd.sort_values("userName").reset_index(drop=True)
+
+        assert_frame_equal(dedup_pd_test, dedup_test)
+
+    def test_shape(self):
+        df = self.build_ak_df()
+
+        row, col = df.shape
+        assert row == 6
+        assert col == 6
+
+    def test_reset_index(self):
+        df = self.build_ak_df()
+
+        slice_df = df[ak.array([1, 3, 5])]
+        assert slice_df.index.to_list() == [1, 3, 5]
+
+        df_reset = slice_df.reset_index()
+        assert df_reset.index.to_list() == [0, 1, 2]
+        assert slice_df.index.to_list(), [1, 3, 5]
+
+        slice_df.reset_index(inplace=True)
+        assert slice_df.index.to_list(), [0, 1, 2]
+
+    def test_rename(self):
+        df = self.build_ak_df()
+
+        rename = {"userName": "name_col", "userID": "user_id"}
+
+        # Test out of Place - column
+        df_rename = df.rename(rename, axis=1)
+        assert "user_id" in df_rename.columns
+        assert "name_col" in df_rename.columns
+        assert "userName" not in df_rename.columns
+        assert "userID" not in df_rename.columns
+        assert "userID" in df.columns
+        assert "userName" in df.columns
+        assert "user_id" not in df.columns
+        assert "name_col" not in df.columns
+
+        # Test in place - column
+        df.rename(column=rename, inplace=True)
+        assert "user_id" in df.columns
+        assert "name_col" in df.columns
+        assert "userName" not in df.columns
+        assert "userID" not in df.columns
+
+        # prep for index renaming
+        rename_idx = {1: 17, 2: 93}
+        conf = list(range(6))
+        conf[1] = 17
+        conf[2] = 93
+
+        # Test out of Place - index
+        df_rename = df.rename(rename_idx)
+        assert df_rename.index.values.to_list() == conf
+        assert df.index.values.to_list() == list(range(6))
+
+        # Test in place - index
+        df.rename(index=rename_idx, inplace=True)
+        assert df.index.values.to_list() == conf
+
+    def test_append(self):
+        df = self.build_ak_df()
+        df_toappend = self.build_ak_append()
+
+        df.append(df_toappend)
```

Comment:
I know I said I would limit stylistic comments, but this var name is really bothering me. It should be `df_toappend`, but since it's only used once we don't need a variable at all

```suggestion
        df.append(self.build_ak_append())
```

There's one other spot in `test_concat ` where `df_toappend` is used that we can also eliminate in a similar fashion

Reply:
Done